log4j-api/src/main/java/org/apache/logging/log4j/util/StackLocator.java

```diff
@@ -77,7 +77,6 @@ public final class StackLocator {
                 }
             }
         } catch (final Exception | LinkageError e) {
-            System.out.println("WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.");
```

Comment:
I would rather keep the warning, but:
 - use `LowLevelLogUtil#log` instead of `System.out.println`,
 - differentiate the message according to `Constants#JAVA_MAJOR_VERSION`:
    - if the version is <= 8 warn about "sun.reflect.Reflection.getCallerClass is not supported. This will impact location-based features.",
    - if the version is > 8 warn about "Runtime environment does not support multi-release JARs. This will impact location-base features."